client/client.go

```diff
@@ -236,6 +244,14 @@ func (c *httpClusterClient) Do(ctx context.Context, act httpAction) (*http.Respo
 		if resp.StatusCode/100 == 5 {
 			continue
 		}
+		if k != pinned {
+			c.Lock()
+			// update pinned if it has not been updated yet
+			if c.pinned == pinned {
```

Comment:
This helps to avoid the pinned keep changing between several values if different Do finds different new pinned value. But we use fixed endpoint loop today, so this is useless now. will delete.